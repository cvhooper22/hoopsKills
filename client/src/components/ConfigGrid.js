import React from 'react';
import PropTypes from 'prop-types';

function ConfigGrid (props) {
  const { data } = props;

  function renderRows () {
    return data.map((row) => {
      return <tr key={ row.id }><td>{ row.label }</td></tr>;
    });
  }

  return (
    <table>
      <tbody>
        { renderRows() }
      </tbody>
    </table>
  );
}

ConfigGrid.propTypes = {
  data: PropTypes.array,
};

ConfigGrid.defaultProps = {
  data: [],
};

export default ConfigGrid;
