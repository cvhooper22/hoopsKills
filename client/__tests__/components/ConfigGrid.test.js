import React from 'react';
import renderer from 'react-test-renderer';
import ConfigGrid from '../../src/components/ConfigGrid';

describe('ConfigGrid -', () => {

  it('Should render empty ConfigGrid', () => {
    const component = renderer.create(<ConfigGrid />);

    expect(component).toMatchSnapshot();
  });

  it('Should render non-empty ConfigGrid', () => {
    const data = [
      { id: 1, label: 'item 1' },
      { id: 2, label: 'item 2' },
      { id: 3, label: 'item 3' },
    ]
    const component = renderer.create(<ConfigGrid data={ data } />);

    expect(component).toMatchSnapshot();
  });
});