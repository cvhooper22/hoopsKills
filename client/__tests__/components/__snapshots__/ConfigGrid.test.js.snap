// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigGrid - Should render empty ConfigGrid 1`] = `
<table>
  <tbody />
</table>
`;

exports[`ConfigGrid - Should render non-empty ConfigGrid 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        item 1
      </td>
    </tr>
    <tr>
      <td>
        item 2
      </td>
    </tr>
    <tr>
      <td>
        item 3
      </td>
    </tr>
  </tbody>
</table>
`;
